io-utils/src/main/java/uk/gov/justice/raml/io/FileTreeScanner.java

```diff
@@ -31,18 +42,35 @@
      * @return a list of paths to matching files under the specified directory
      */
     public Collection<Path> find(final Path baseDir, final String[] includes, final String[] excludes) throws IOException {
+        if (!shouldSearchOnClasspath(baseDir) && !baseDir.toFile().exists()) {
+            return emptyList();
+        }
 
-        LOGGER.debug(format("Finding files in: %s", baseDir.toString()));
+        Reflections reflections = new Reflections(
+                new ConfigurationBuilder()
+                        .filterInputsBy(filterOf(includes, excludes))
+                        .setUrls(shouldSearchOnClasspath(baseDir)
+                                ? ClasspathHelper.forClassLoader()
+                                : asList(baseDir.toUri().toURL()))
```

Comment:
Inline if might be clearer if moved to a variable above this builder statement, which is already quite complex to read